### PR TITLE
[WIP] Ubuntu has various createrepo dependencies depending on its version…

### PR DESCRIPTION
**Pull request by gounthar**
This is linked to issue #314.

Ubuntu had [`createrepo`](https://answers.launchpad.net/createrepo/+question/690448) until `18.04`, then it disappeared in `20.04`, and
reappeared in `21.04` as `createrepo-c` which is a rewrite in `C`.
We should then choose `createrepo` or `createrepo-c` or nothing,
depending on the `Ubuntu` distro version.

Here is a rough draft of `setup.sh` that chooses `createrepo` or `createrepo-c` depending on the `Ubuntu` version.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by basil** [comment]
> Apparently trying to bring a solution to a problem that doesn't exist.

In #314 I wrote that the high-level problem was unclear to me, not that it did not exist.

If the problem is that one dislikes the technical debt of depending on older versions of Ubuntu and wants to upgrade our build toolchain, then a PR like this would indeed be needed, along with corresponding changes to the Docker image as in https://github.com/jenkins-infra/docker-packaging/pull/28 and its associated prerequisite infrastructure work. I would welcome such a contribution!

But in https://github.com/jenkinsci/packaging/issues/314#issuecomment-1145194091 you wrote that your problem was "trying to build and test some changes for the `armhf` platform" locally.

In other words, this PR is of value, but it seems to solve a _different_ (not non-existent) high-level problem than the one you actually have.

**Comment by timja** [comment]
FWIW https://github.com/jenkins-infra/docker-packaging/pull/33 is merged, and a follow up of https://github.com/jenkins-infra/docker-packaging/pull/34 is open
